tls/extensions/s2n_server_key_share.c

```diff
@@ -45,6 +46,27 @@ int s2n_extensions_server_key_share_send_check(struct s2n_connection *conn)
 
     return 0;
 }
+/*
+ * Selects highest priority mutually supported keyshare
+ */
+int s2n_extensions_server_key_share_select(struct s2n_connection *conn)
+{
+    for (uint32_t i = 0; i < s2n_ecc_evp_supported_curves_list_len; i++) {
+        /* Checks supported group and keyshare have both been sent */
+        if (conn->secure.client_ecc_evp_params[i].negotiated_curve &&
+             conn->secure.mutually_supported_groups[i]) {
+            conn->secure.server_ecc_evp_params.negotiated_curve = conn->secure.client_ecc_evp_params[i].negotiated_curve;
+            return 0;
+        }
+    }
+    /* Client sent no keyshares, need to send Hello Retry Request with first negotiated curve */
+    if (conn->secure.server_ecc_evp_params.negotiated_curve) {
+        /* Commented out until retry is implemented in issue #1607.
+        GUARD(s2n_server_should_retry(conn)); */
```

Comment:
What happens here today? An error further down the  chain?

Reply:
It would cause an error in this function: s2n_extensions_server_key_share_send_check